cmd/checkversions/checkversions.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+#set -x
+
+#debug="--verbose --debug"
+
+runtime_repo="github.com/kata-containers/runtime"
+runtime_repo_dir="$GOPATH/src/${runtime_repo}"
+versions_file="${runtime_repo_dir}/versions.yaml"
+YQ=$(which yq)
+uscan_phrase="uscan-url"
+
+
+debug() {
+	# echo $*
+	true
+}
+
+# Look in the yaml file handed to us to find uscan URLs.
+# return a list of paths to the URL items.
+extract_uscan_items() {
+	vfile=$1
+	local tmpjson=$(mktemp checkverXXX.json)
+	
+	${YQ} r -j "$1" > "$tmpjson"
+	paths=$(jq -r 'paths | join(".")' $tmpjson | grep "${uscan_phrase}$")
+
+	rm $tmpjson
+	echo "$paths"
+}
+
+main() {
+	local fails=0
+	local passes=0
+
+	local paths=$(extract_uscan_items "$versions_file")
+
+	debug "Found [$paths]"
+
+	for path in $paths; do
+		debug ""
+		local rootname="${path%.$uscan_phrase}"
+		echo "Processing [$rootname]"
+		local versionpath="$rootname.version"
+		debug "Version path [$versionpath]"
+		version=$(yq r $versions_file $versionpath)
+		debug "Version [$version]"
+		uscanurl=$(yq r $versions_file $path)
+		debug "uscan URL is [$uscanurl]"
+		local optspath="$rootname.uscan-opts"
+		uscanopts=$(yq r $versions_file $optspath)
+		debug "uscan OPTS is [$uscanopts]"
+
+		debug "Generating watchfile"
+		echo "version=4" > watchfile
+		if [ -n "$uscanopts" ] && [ "$uscanopts" != "null" ]; then
+			echo "$uscanopts  \\" >> watchfile
+		fi
+		echo "$uscanurl" >> watchfile
+		result="$(uscan --report --package ${rootname} --upstream-version ${version} --watchfile watchfile 2>/dev/null)"
+
+		failure=$(grep Newer <<< $result)
+
+		if [ -n "$failure" ]; then
+			echo "----------------------"
+			echo "FAILED: $path"
+			echo "$result"
+			echo "----------------------"
+			echo
+			((fails++))
+		else
+			echo "----------------------"
+			echo "PASSED: $path"
+			echo "$result"
+			echo "----------------------"
+			echo
+			((passes++))
+		fi
+	done
+
+	echo "PASSES: $passes"
+	echo "FAILS: $fails"
```

Comment:
How about adding the following to allow the script to return a useful value:

```
exit "$fails"
```

Reply:
ack, good idea.